src/olympia/versions/tests/test_models.py

```diff
@@ -1118,6 +1118,32 @@ def test_version_kept_when_license_deleted(self):
         assert not License.objects.filter(pk=license.pk).exists()
         assert Version.objects.filter(pk=self.version.pk).exists()
 
+    def test_has_been_human_reviewed(self):
+        assert AutoApprovalSummary.objects.count() == 0
+        self.version.file.update(status=amo.STATUS_DISABLED)
+        assert not self.version.has_been_human_reviewed
+
+        self.version.file.update(reviewed=datetime.now())
+        assert self.version.has_been_human_reviewed
+
+        self.version.file.update(status=amo.STATUS_NOMINATED)
+        assert not self.version.has_been_human_reviewed
+
+        self.version.file.update(status=amo.STATUS_APPROVED)
+        assert self.version.has_been_human_reviewed
+
+        summary = AutoApprovalSummary.objects.create(version=self.version)
+        assert self.version.has_been_human_reviewed
+
+        summary.update(verdict=amo.AUTO_APPROVED)
+        assert not self.version.has_been_human_reviewed
+
+        summary.update(verdict=amo.NOT_AUTO_APPROVED)
+        assert self.version.has_been_human_reviewed
+
+        summary.update(verdict=amo.AUTO_APPROVED, confirmed=True)
+        assert self.version.has_been_human_reviewed
+
```

Comment:
https://github.com/mozilla/addons-server/pull/19384/commits/3ccb1ac4d2c51e7ec7d05243e710ce2b6242a662 but I'm not super convinced that all the combinations of states are covered.  It's also a little inconsistent that only `STATUS_DISABLED` looks at `reviewed` but I couldn't think of another way of differentiating rejected from disabled-because-it-was-superseded.